hackathon_site/hardware/serializers.py

```diff
@@ -89,6 +89,56 @@ def get_team_code(obj: Order):
         return obj.team.team_code
 
 
+class OrderChangeSerializer(serializers.ModelSerializer):
+    hardware = HardwareSerializer(many=True, read_only=True)
+    team_code = serializers.SerializerMethodField()
+
+    change_options = {
+        "Submitted": ["Cancelled", "Ready for Pickup"],
+        "Ready for Pickup": ["Picked Up"],
+    }
+
+    class Meta:
+        model = Order
+        fields = OrderListSerializer.Meta.fields
+        read_only_fields = (
+            "id",
+            "hardware",
+            "team",
+            "team_code",
+            "created_at",
+            "updated_at",
+        )
+
+    @staticmethod
+    def get_team_code(obj: Order):
+        return obj.team.team_code
+
+    def validate(self, data):
+        current_status = self.instance.status
+
+        if current_status not in self.change_options:
+            raise ValidationError(
+                {"detail": "Cannot change the status for this order."},
+                code=status.HTTP_403_FORBIDDEN,
```

Comment:
I think this one should still be a 400? That'll just make the logic on the frontend a bit easier, since we can generically catch 400s and display them in snackbars

Reply:
Also if you want a 400 status code, you don't need to specify it (that's the default)